actionview/lib/action_view/template_details.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module ActionView
+  class TemplateDetails # :nodoc:
+    class Requested
+      attr_reader :locale, :handlers, :formats, :variants
+
+      def initialize(locale:, handlers:, formats:, variants:)
+        @locale = locale
+        @handlers = handlers
+        @formats = formats
+        @variants = variants
+      end
+    end
+
+    attr_reader :locale, :handler, :format, :variant
```

Comment:
Is it important that there are two sets of `attr_reader` declarations, with one accepting singular names, one accepting plural names, and `locale` in both?

Reply:
These are separate classes. The way details work is that the way they're requested in, as arrays of acceptable values, is different than how they're represented on the template, as singular values

For some reason, though all the other keys in the details hash were pluralized, `locale` has never been https://github.com/rails/rails/blob/main/actionview/lib/action_view/lookup_context.rb#L42-L48. We considered using `locales` here, but decided to prefer one kind of consistency over another.